server/src/bge/System.cpp

```diff
@@ -593,4 +593,83 @@ namespace bge {
             }
         }
     }
+
+    // ------------------------------------------------------------------------------------------------------------------------------------------------
+
+    SeasonEffectSystem::SeasonEffectSystem(
+        World* _gameWorld,
+		std::shared_ptr<ComponentManager<HealthComponent>> _healthCM,
+		std::shared_ptr<ComponentManager<VelocityComponent>> _velocityCM,
+        std::shared_ptr<ComponentManager<MovementRequestComponent>> _movementRequestCM, 
+        std::shared_ptr<ComponentManager<JumpInfoComponent>> _jumpInfoCM, 
+        std::shared_ptr<ComponentManager<SeasonAbilityStatusComponent>> _seasonAbilityStatusCM) {
+        
+        world = _gameWorld;
+        healthCM = _healthCM;
+        velocityCM = _velocityCM;
+        movementRequestCM = _movementRequestCM;
+        jumpInfoCM = _jumpInfoCM;
+        seasonAbilityStatusCM = _seasonAbilityStatusCM;
+        counter = 0;
+
+    }
+
+    void SeasonEffectSystem::update() {
+
+        // Change this to make each season longer or shorter
+        if (counter > 500) {
+            counter = 0;
+            world->currentSeason = (world->currentSeason+1)%4;
+            // std::printf("Current Season is %d\n", world->currentSeason);
+        }
+
+        if (world->currentSeason == SPRING_SEASON) {
+            for (Entity e : registeredEntities) {
+                HealthComponent& health = healthCM->lookup(e);
+                if (counter % 50 == 0) {
+                    health.healthPoint = std::min(100,health.healthPoint+5);
+                }
+            }
+        } else if (world->currentSeason == SUMMER_SEASON) {
+            for (Entity e : registeredEntities) {
+                JumpInfoComponent& jump = jumpInfoCM->lookup(e);
+                MovementRequestComponent& req = movementRequestCM->lookup(e);
+                VelocityComponent& vel = velocityCM->lookup(e);
+
+                if (!jump.jumpHeld && req.jumpRequested && jump.doubleJumpUsed == MAX_JUMPS_ALLOWED) {
+                    jump.doubleJumpUsed++;
+                    vel.velocity.y = JUMP_SPEED*1.25;
+                    jump.jumpHeld = true;
+                }
+            }
+        } else if (world->currentSeason == AUTUMN_SEASON) {
+            for (Entity e : registeredEntities) {
+                SeasonAbilityStatusComponent& seasonAbilityStatus = seasonAbilityStatusCM->lookup(e);
+                // Reduce cooldown by 66%
+                if (seasonAbilityStatus.coolDown < SEASON_ABILITY_CD*2/3) {
+                    seasonAbilityStatus.coolDown = 0;
+                }
+            }
+        } else if (world->currentSeason == WINTER_SEASON) {
+            for (Entity e : registeredEntities) {
+                MovementRequestComponent& req = movementRequestCM->lookup(e);
+                VelocityComponent& vel = velocityCM->lookup(e);
+                if (!req.jumpRequested) {
```

Comment:
Is the idea to make the player speed up while on the ground or make the player speed up while not trying to jump? The former sounds more like sliding on ice to me, but I think the latter is what's implemented here.

Reply:
prolly use vel.onGround here?

Reply:
That was the initial implementation, I think I changed it to test something. I changed it back.